dbt_subprojects/dex/macros/models/_project/uniswap_downstream_trades.sql

```diff
@@ -0,0 +1,371 @@
+{% macro uniswap_downstream_trades(
+    blockchain = null
+    , has_univ4 = null 
+    , has_bunni = null 
+    )
+%}
+
+with 
+
+v4_trades as (
+    {% if has_univ4|lower == 'yes' %}
+    select
+        blockchain
+        , tx_hash 
+        , evt_index 
+        , fee 
+        , block_time 
+        , block_date 
+        , block_number 
+    from
+    {{ ref('uniswap_v4_' + blockchain + '_base_trades') }}
+    where 1 = 1 
+    {% if is_incremental() %}
+    and {{ incremental_predicate('block_time') }}
+    {% endif %}
+    {% else %}
+    select 
+        cast(null as varchar) as blockchain 
+        , cast(null as varbinary) as tx_hash 
+        , cast(null as bigint) as evt_index 
+        , cast(null as double) as fee 
+        , cast(null as timestamp) as block_time 
+        , cast(null as timestamp) as block_date 
+        , cast(null as bigint) as block_number 
+    where false
+    {% endif %}
+
+),
+
+bunni_fees as (
+    {% if has_bunni|lower == 'yes' %}
+    select
+        '{{blockchain}}' as blockchain
+        , evt_tx_hash  as tx_hash
+        , fee  
+        , evt_block_date as block_date
+        , id 
+        , evt_index + 1 as evt_index
+        , 'bunni' as hooks 
+    from
+    {{ source('bunni_v2_' + blockchain, 'bunnihook_evt_swap') }}
+    where 1 = 1 
+    {% if is_incremental() %}
+    and {{ incremental_predicate('evt_block_time') }}
+    {% endif %}
+    {% else %}
+    select 
+        cast(null as varchar) as blockchain 
+        , cast(null as varbinary) as tx_hash 
+        , cast(null as double) as fee 
+        , cast(null as timestamp) as block_date 
+        , cast(null as varbinary) as id
+        , cast(null as bigint) as evt_index 
+        , cast(null as varchar) as hooks 
+    where false 
+    {% endif %}
+),
+
+flaunch_hookswap as (
+    select 
+        * 
+    from 
+    {{ source('flaunch_base', 'positionmanager_v2_evt_hookswap') }} 
```

Comment:
is the `_base` here in schema referring to blockchain? are these all `base` only? will that affect data downstream for other chains?

Reply:
yeah it's referring to blockchain so these are all base only and wouldn't affect data downstream for other chains